inference-engine/thirdparty/clDNN/tests/test_cases/reorder_gpu_test.cpp

```diff
@@ -2396,7 +2398,207 @@ INSTANTIATE_TEST_SUITE_P(DISABLED_REORDER,
+            VF<int32_t> rnd_vec = generate_random_1d<int32_t>(s.count() / 32, min_random, max_random);
+            set_values(prim, rnd_vec);
+        } else if (l.data_type == data_types::i8 || l.data_type == data_types::u8) {
+            VF<uint8_t> rnd_vec = generate_random_1d<uint8_t>(s.count(), min_random, max_random);
+            set_values(prim, rnd_vec);
+        } else if (l.data_type == data_types::f16) {
+            VF<uint16_t> rnd_vec = generate_random_1d<uint16_t>(s.count(), -1, 1);
+            set_values(prim, rnd_vec);
+        } else {
+            VF<float> rnd_vec = generate_random_1d<float>(s.count(), -1, 1);
+            set_values(prim, rnd_vec);
+        }
+
+        return prim;
+    }
+
+    layout get_input_layout(T& p) {
+        auto pad = p.pad;
+        std::vector<int> pad_ = { 0, 0, pad.spatial[0], pad.spatial[1] };
+        return layout{ p.data_type, p.input_format, p.in_shape, padding{pad_} };
+    }
+
+    layout get_weights_layout(T& p, const int32_t /* split */ = 1) {
+        cldnn::tensor weights_tensor;
+        weights_tensor = cldnn::tensor(batch(p.out_shape.feature[0]), feature(p.in_shape.feature[0]), spatial(p.kernel.spatial[0], p.kernel.spatial[1], p.kernel.spatial[2]));
+        return layout{p.weights_type, p.weights_format, weights_tensor};
+    }
+
+    layout get_bias_layout(T& p) {
+        return layout{ p.default_type, format::bfyx, tensor{1, p.out_shape.feature[0], 1, 1} };
+    }
+
+    template <class... Args>
+    void create_topologies(Args const&... args) {
+        topology_test.add(args...);
+    }
+};
+
+class testing_removal_reorder : public ReorderTest<reorder_test_param> {};
+TEST_P(testing_removal_reorder, not_to_remove_reorder_wide_depth) {
+    auto p = GetParam();
+    // layout reorder_layout(data_types::u8, format::b_fs_yx_fsv32, p.in_shape, padding({0, 0, 1, 1}, 0));
+    layout reorder_layout(data_types::u8, format::b_fs_yx_fsv32, p.in_shape, padding({0, }, 0));
+
+    create_topologies(input_layout("input", get_input_layout(p)),
+        data("weights", get_mem(get_weights_layout(p))),
+        data("bias", get_mem(get_bias_layout(p))),
+        data("weights_sec", get_mem(get_weights_layout(p))),
+        reorder("reorder_fp32", "input", format::bfyx, data_types::f32),
+        convolution("conv_prim", "reorder_fp32", { "weights" }, { "bias" }, 1, p.stride, p.pad, tensor{1}, p.in_shape, data_types::u8, false),
+        reorder("reorder_conv", "conv_prim", reorder_layout),
+        convolution("conv_output", "reorder_conv", { "weights_sec" }, 1, p.stride, p.pad),
+        reorder("reorder_bfyx", "conv_output", format::b_fs_yx_fsv32, data_types::f32),
+        resample("resample", "reorder_bfyx", p.out_shape, 1),
+        reorder("reorder_output", "resample", p.default_format, data_types::f32)
+    );
+
+    execute(p);
+
+    EXPECT_EQ(check_not_removed(p, "reorder_conv"), true);
+}
+
+TEST_P(testing_removal_reorder, removal_metaonly_reorder) {
```

Comment:
This test succeeds even though your new code in `are_layouts_identical` is removed. I don't think that is your intention here. Could you review it?

Reply:
It is not intended. There was a removed reoder between conv and eltwise. That is the root cause to make difference.

Reply:
Fixed